mesa_expense_autopopulate/models/hr/expense.py

```diff
@@ -0,0 +1,160 @@
+    force_project_id = fields.Many2one("project.project", store=True)
+    force_team_id = fields.Many2one("crm.team", store=True)
+    force_user_id = fields.Many2one("res.users", store=True)
+    force_warehouse_id = fields.Many2one("stock.warehouse", store=True)
+
+    # The following studio fields are only included here because the database
+    # would otherwise not let this module install.
+    x_studio_aa_update = fields.Many2one("account.analytic.account")
+    x_studio_field_Evfct = fields.Many2one("project.project")
+
+    @api.multi
+    @api.depends("sale_order_id.x_studio_project")
+    def _compute_project(self):
+        """
+        Compute the Expense's Project. It defaults to the Expense's Sale Order's
+        Project.
+
+        :return {NoneType}:
+        """
+        for expense_id in self:
+            expense_id.project_id = (
+                expense_id.force_project_id.id or expense_id.sale_order_id.project_id
+            )
+
+    @api.multi
+    @api.depends("sale_order_id.team_id")
+    def _compute_team(self):
+        """
+        Compute the Expense's Sales Team. It defaults to the Expense's Sale
+        Order's Sales Team.
+
+        :return {NoneType}:
+        """
+        for expense_id in self:
+            expense_id.team_id = (
+                expense_id.force_team_id.id or expense_id.sale_order_id.team_id.id
+            )
+
+    @api.multi
+    @api.depends("sale_order_id.warehouse_id")
+    def _compute_warehouse(self):
+        """
+        Compute the Expense's Warehouse. It defaults to the Expense's Sale
+        Order's Warehouse.
+
+        :return {NoneType}:
+        """
+        for expense_id in self:
+            expense_id.warehouse_id = (
+                expense_id.warehouse_id.id or expense_id.sale_order_id.warehouse_id.id
+            )
+
+    @api.multi
+    @api.depends("sale_order_id.user_id")
+    def _compute_user(self):
+        """
+        Compute the Expense's Salesperson. It defaults to the Expense's Sale
+        Order's Salesperson.
+
+        :return {NoneType}:
+        """
+        for expense_id in self:
+            expense_id.user_id = (
+                expense_id.force_user_id.id or expense_id.sale_order_id.user_id.id
+            )
+
+    @api.multi
+    @api.depends("sale_order_id.company_id")
+    def _compute_company(self):
+        """
+        Compute the Expense's Company. It defaults to the Expense's Sale Order's
+        Company.
+
+        :return {NoneType}:
+        """
+        for expense_id in self:
+            expense_id.company_id = (
+                expense_id.force_company_id.id or expense_id.sale_order_id.company_id.id
+            )
+
+    @api.multi
+    def _inverse_project(self):
+        """
+        This allows users to manually set the Project field in the form.
+
+        :return {NoneType}:
+        """
+        for expense_id in self:
+            expense_id.force_project_id = expense_id.x_studio_field_Evfct.id
```

Comment:
What is going on here?
```suggestion
            expense_id.force_project_id = expense_id.project_id.id
```